projects/soft_999_shicheng_lu_fw_103/src/main.c

```diff
@@ -0,0 +1,61 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "interrupt_def.h"
+#include "log.h"
+#include "wait.h"
+
+// LOG_DEBUG the adc converted value of context pin when called (context is A6 pin)
+static void prv_log_adc_value(const GpioAddress *address, void *context) {
+  GpioAddress *adc_addr = context;
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(*adc_addr, &adc_data);

```

Comment:
This is good. However, notice that this function returns a value of type `StatusCode` which tells us if the function works properly. What happens if this function fails and returns a failure rather than OK. 

Right now we aren't checking for it, so if we got bad data from the function we would not know and we would be printing it out still. It would be best to add a check for this.